arches_for_science/media/js/utils/iiif-utils.js

```diff
@@ -0,0 +1,158 @@
+define([], function() {
+    const getVersion = (manifestData) => {
+        const urlString = manifestData?.['@context'];
+        const url = new URL(urlString);
+        if (url.pathname.split("/")[3].startsWith("3")) {
+            return 3;
+        } else if (url.pathname.split("/")[3].startsWith("2")) {
+            return 2;
+        }
+    };
+    const getManifestDataValue = function(object, property, returnFirstVal, version, i18n) {
+        let val = object[property];
+        if (version === 2) {
+            if (Array.isArray(val) && returnFirstVal) val = val[0];
+        } else if (version === 3) {
+            if (i18n) val = val[i18n];
+            if (Array.isArray(val) && returnFirstVal) val = val[0];
+        }
+        return val;
+    };
+    const getCanvases = (manifestData) => {
+        const canvases = [];
+        if (getVersion(manifestData) === 3) {
```

Comment:
It's probably just a bit better to check the version once at the top of the function rather than in each conditional statement.
```suggestion
        const version = getVersion(manifestData)
        if (version === 3) {
```